### PR TITLE
ceph: added rook discover devices interval to helm charts

### PR DESCRIPTION
**Pull request by jaxels10**
On request from Alexander Trost, I have added the ROOK_DISCOVER_DEVICES_INTERVAL variable to cluster/charts/rook-ceph/values.yaml and to cluster/charts/rook-ceph/templates/deployment.yaml.

Signed-off-by: Jesper Axelsen <jesperbaxelsen@gmail.com>

**Description of your changes:
On request from Alexander Trost, I have added the ROOK_DISCOVER_DEVICES_INTERVAL variable to cluster/charts/rook-ceph/values.yaml and to cluster/charts/rook-ceph/templates/deployment.yaml.**

**Which issue is resolved by this Pull Request: Ability to use ROOK_DISCOVER_DEVICES_INTERVAL in the ceph helm charts**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: cluster/charts/rook-ceph/values.yaml [hunk 1]
```diff
@@ -332,6 +332,9 @@ allowMultipleFilesystems: false
 # For more details see https://github.com/rook/rook/issues/2417
 enableSelinuxRelabeling: true
 
+# The duration between discovering devices in the rook-discover daemonset.
+discoverDevicesInterval: 60m
```
Comment (Madhu-1) [inline cluster/charts/rook-ceph/values.yaml:336]: this needs to be documented at https://github.com/rook/rook/blob/master/Documentation/helm-operator.md#configuration and do we need to set it by 60m by default?

File: cluster/charts/rook-ceph/values.yaml [hunk 2]
```diff
@@ -357,3 +360,4 @@ admissionController:
   #      operator: Exists
   #      effect: NoSchedule
   # nodeAffinity: key1=value1,value2; key2=value3
+
```
Comment (Madhu-1) [inline cluster/charts/rook-ceph/values.yaml:363]: the extra line is not required here?

Comment (jaxels10) [reply cluster/charts/rook-ceph/values.yaml:363]: True, that was unintentionally added

**Comment by jaxels10** [comment]
Sorry for the mess I have been creating this morning, I am quite new with this. I have tried to clean up best I could, but let me know if I need to do anything else

File: cluster/charts/rook-ceph/values.yaml (hunk 1 above)
Comment (galexrt) [reply cluster/charts/rook-ceph/values.yaml:336]: @Madhu-1 The "default" is `60m`, see https://github.com/rook/rook/blob/2bb94441304857870ab81f5fa7b324695f1068f1/pkg/operator/discover/discover.go#L55

Comment (galexrt) [inline cluster/charts/rook-ceph/values.yaml:336]: I'm a bit conflicted about having the `discoverDevicesInterval` not in the `discover:` section above (line 320), but here.

What do you think?

Comment (jaxels10) [reply cluster/charts/rook-ceph/values.yaml:336]: I don't really have a preference with that, I just wasn't sure if it belonged there. Want me to change it?

Comment (BlaineEXE) [reply cluster/charts/rook-ceph/values.yaml:336]: I think it belongs in the `discover:` section.

**Comment by jaxels10** [comment]
@travisn I will look at that right away, thanks for that!

File: cluster/charts/rook-ceph/values.yaml (hunk 1 above)
Comment (jaxels10) [reply cluster/charts/rook-ceph/values.yaml]: @BlaineEXE I have now added it to the discover: section. Thanks for the input

**Review by travisn — APPROVED** [review]
The changes look good, we can go ahead and merge. 
@jaxels10 Did you find the hotplug settings helpful as well?

**Review by travisn — CHANGES_REQUESTED** [review]
Actually, if you could just squash the commits before we merge, thanks!

**Comment by jaxels10** [comment]
> Actually, if you could just squash the commits before we merge, thanks!

@travisn should be squashed now, but I see another commit 9256b56 on this branch. Should this be squashed into it as well?

**Comment by travisn** [comment]
> > Actually, if you could just squash the commits before we merge, thanks!
> 
> @travisn should be squashed now, but I see another commit [9256b56](https://github.com/rook/rook/commit/9256b56f8fcfbaa6b696b2e7bdc173acd23a6ac1) on this branch. Should this be squashed into it as well?

Looks like something didn't quite go right with the squash. There should only be a single commit in the PR after the squash. How did you squash? Maybe this will help: `git rebase upstream/master`

**Comment by leseb** [comment]
Closing due to inactivity and review were never addressed. Feel free to re-open, thanks.

**Pull request closed without merging**